mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -131,10 +132,14 @@ getObject' =
 
 read :: Address -> AWS (Maybe Text)
 read a = do
-  resp <- getObject' a
-  let format y = T.concat . TL.toChunks . TL.decodeUtf8 $ y
-  z <- liftIO . sequence $ (\r -> runResourceT (r ^. gorsBody . to bodyResponse $$+- sinkLbs)) <$> resp
-  pure (format <$> z)
+  r <- read' a
+  z <- liftIO . sequence $ (runResourceT . ($$+- sinkLbs)) <$> r
+  pure $ fmap (T.concat . TL.toChunks . TL.decodeUtf8) z
+
+read' :: Address -> AWS (Maybe (ResumableSource (ResourceT IO) BS.ByteString))
```

Comment:
I wonder what would happen if a test for new `read'` never consumes the result or consumes it only partially. It is probably all handled by `conduit` but still would be interesting to see how it works with AWS...